src/main/scala-2.11/markets/engines/CDAMatchingEngine.scala

```diff
@@ -82,6 +111,42 @@ class CDAMatchingEngine(askOrdering: PriceOrdering[AskOrder],
     math.min(incoming.quantity, existingOrder.quantity)
   }
 
+  /** Remove an order from the matching engine. */
+  def remove(existing: Order): Option[Order] = {
+    existing match {
+      case _: AskOrder =>
+        _askOrderBook.find(o => o.uuid == existing.uuid) match {
+          case result @ Some(residualOrder) =>
+            _askOrderBook.remove(residualOrder) // SIDE EFFECT!
+            result
+          case _ => None
+        }
+      case _: BidOrder =>
+        _bidOrderBook.find(o => o.uuid == existing.uuid) match {
+          case result @ Some(residualOrder) =>
+            _bidOrderBook.remove(residualOrder) // SIDE EFFECT!
+            result
+          case _ => None
+        }
+    }
+  }
+
+  /* Mutable collection of ask orders for internal use only! */
+  protected val _askOrderBook = mutable.TreeSet.empty[AskOrder](askOrdering)
+
+  /* Mutable collection of bid orders for internal use only! */
+  protected val _bidOrderBook = mutable.TreeSet.empty[BidOrder](bidOrdering)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [It's preferred to use immutable collections instead of mutable ones](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1834547231/issues/source?bid=2921361&fileBranchId=3080901#l138)